scylla/src/transport/session.rs

```diff
@@ -280,18 +280,17 @@ impl Session {
         query: impl Into<Query>,
         values: impl ValueList,
     ) -> Result<Option<Vec<result::Row>>, QueryError> {
-        let query: Query = query.into();
-        let query_text: &str = query.get_contents();
+        let query = query.into();
+        let query_text = query.get_contents();
 
         // In case the user tried doing session.query("use keyspace ks") run session::use_keyspace
-        if query_text.to_lowercase().starts_with("use ") {
+        if query_is_setting_keyspace(query_text) {
```

Comment:
I think it's cleaner to do this in a separate method (testing is easier that way)